src/main/java/com/faforever/api/data/domain/MapVersion.java

```diff
@@ -1,166 +1,106 @@
 package com.faforever.api.data.domain;
 
 import com.yahoo.elide.annotation.Include;
+import lombok.Setter;
 
-import javax.persistence.Basic;
 import javax.persistence.Column;
 import javax.persistence.Entity;
+import javax.persistence.GeneratedValue;
+import javax.persistence.GenerationType;
 import javax.persistence.Id;
+import javax.persistence.JoinColumn;
+import javax.persistence.ManyToOne;
 import javax.persistence.Table;
+import javax.validation.constraints.NotNull;
 import java.sql.Timestamp;
-import java.util.Objects;
 
 @Entity
+@Setter
 @Table(name = "map_version")
 @Include(rootLevel = true, type = "map_version")
 public class MapVersion {
 
-  private Integer id;
+  private int id;
   private String description;
   private Integer maxPlayers;
   private int width;
   private int height;
   private int version;
   private String filename;
-  private byte ranked;
-  private byte hidden;
+  private boolean ranked;
+  private boolean hidden;
   private Timestamp createTime;
   private Timestamp updateTime;
+  private Map map;
 
   @Id
+  @GeneratedValue(strategy = GenerationType.IDENTITY)
   @Column(name = "id")
-  public Integer getId() {
+  public int getId() {
     return id;
   }
 
-  public void setId(Integer id) {
-    this.id = id;
-  }
-
-  @Basic
   @Column(name = "description")
   public String getDescription() {
     return description;
   }
 
-  public void setDescription(String description) {
-    this.description = description;
-  }
-
-  @Basic
   @Column(name = "max_players")
+  @NotNull
   public Integer getMaxPlayers() {
     return maxPlayers;
   }
 
-  public void setMaxPlayers(Integer maxPlayers) {
-    this.maxPlayers = maxPlayers;
-  }
-
-  @Basic
   @Column(name = "width")
+  @NotNull
   public int getWidth() {
     return width;
   }
 
-  public void setWidth(int width) {
-    this.width = width;
-  }
-
-  @Basic
   @Column(name = "height")
+  @NotNull
```

Comment:
You are annotating primitives with `@NotNull` - they can never be null per definition.